File: kotlin/FindDuplicatePrintOnce.kt
fun findAndPrintDuplicates(arr: IntArray) {
    val seen = mutableSetOf<Int>()
    val duplicates = mutableSetOf<Int>()
    
    // Find duplicates
    for (num in arr) {
        if (num in seen) {
            duplicates.add(num)
        } else {
            seen.add(num)
        }
    }
    
    // Print duplicates (automatically printed once due to Set)
    if (duplicates.isEmpty()) {
        println("No duplicates found.")
    } else {
        duplicates.forEach { println(it) }
    }
}

fun main() {
    // Sample array with duplicates
    val numbers = intArrayOf(1, 2, 3, 4, 2, 5, 6, 3, 7, 8, 1, 9, 4)
    
    println("Original Array: ${numbers.contentToString()}")
    println("\nDuplicate elements (printed once):")
    
    findAndPrintDuplicates(numbers)
}

File: csharp/FindDuplicatePrintOnce.cs
using System;
using System.Collections.Generic;
using System.Linq;

class FindDuplicatePrintOnce
{
    static void Main(string[] args)
    {
        // Sample array with duplicates
        int[] numbers = { 1, 2, 3, 4, 2, 5, 6, 3, 7, 8, 1, 9, 4 };
        
        Console.WriteLine("Original Array: [" + string.Join(", ", numbers) + "]");
        Console.WriteLine("\nDuplicate elements (printed once):");
        
        FindAndPrintDuplicates(numbers);
    }
    
    static void FindAndPrintDuplicates(int[] arr)
    {
        HashSet<int> seen = new HashSet<int>();
        HashSet<int> duplicates = new HashSet<int>();
        
        // Find duplicates
        foreach (int num in arr)
        {
            if (seen.Contains(num))
            {
                duplicates.Add(num);
            }
            else
            {
                seen.Add(num);
            }
        }
        
        // Print duplicates (automatically printed once due to HashSet)
        if (duplicates.Count == 0)
        {
            Console.WriteLine("No duplicates found.");
        }
        else
        {
            foreach (int duplicate in duplicates)
            {
                Console.WriteLine(duplicate);
            }
        }
    }
}
